### PR TITLE
[0.2.dev5] Template for column from expression

### PR DESCRIPTION
**Pull request by smmaurer**
This PR adds a template to register a column of derived data with Orca, based on an expression. This is the first use case from issue #98.

### New template

- `urbansim_templates.data.ColumnFromExpression()`

The expression is passed to `df.eval()` and can refer to other columns in the same table, including stand-alone columns that have not yet been evaluated. It supports arithmetic and math functions including sqrt, abs, log, exp, log1p, and expm1.

More info about the expression parsing:
- [Pandas df.eval()](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.eval.html)
- [Pandas discussion of eval() implementation](https://pandas.pydata.org/pandas-docs/stable/user_guide/enhancingperf.html#enhancingperf-eval)
- [NumExpr](https://numexpr.readthedocs.io/en/latest/), the underlying engine

Template-specific parameters:
- `column_name`
- `table`
- `expression`
- `data_type` (optional)
- `missing_values` (optional)
- `cache` (optional)
- `cache_scope` (optional)

### Versioning

- 0.2.dev5

### To do before merging

- [x] finish tests
- [x] add check for column validity in `run()` method
- [x] finalize versioning
- [x] update docs and changelog

### CODE REVIEW

File: urbansim_templates/data/column_from_expression.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        
+        Requires values to be set for ``column_name``, ``table``, and ``expression``.
+        
+        Returns
+        -------
+        None
+        
+        """
+        if self.column_name is None:
+            raise ValueError("Please provide a column name")
```
Comment (apdjustino) [inline urbansim_templates/data/column_from_expression.py]: This line uses double quotes but there are single quotes in other parts of the file. In the future we can use pylint to ensure uniform code style.

File: urbansim_templates/data/column_from_expression.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        
+        if self.table is None:
+            raise ValueError("Please provide a table")
```
Comment (apdjustino) [inline urbansim_templates/data/column_from_expression.py]: Same as above

File: urbansim_templates/data/column_from_expression.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        
+        if self.expression is None:
+            raise ValueError("Please provide an expression")
```
Comment (apdjustino) [inline urbansim_templates/data/column_from_expression.py]: Same as above

File: urbansim_templates/data/column_from_expression.py (hunk 1 above)
Comment (smmaurer) [reply urbansim_templates/data/column_from_expression.py]: I use double quotes for natural-language messages and single quotes for strings that are more like data, i think. Not sure if this is a standard or just a habit i picked up, though! We can see what pylint wants.

**Pull request merged**